ignoreNotFound operator (#1)
nilsmehlhorn/ngx-operators: README.md

```

[__read more__](https://nils-mehlhorn.de/posts/angular-file-download-progress)

### optional

> Maps 404 error responses to `undefined`; effectively marking the resource as optional.

Reviewer: ```suggestion
Ignores 404 error responses by instead completing the underlying observable.
```

```

[__read more__](https://nils-mehlhorn.de/posts/angular-file-download-progress)

### ignoreNotFound

> Ignores 404 error responses by instead completing the underlying observable.

Reviewer: ```suggestion
Ignores 404 error responses by instead completing the underlying observable.

Note: You can use [defaultIfEmpty](https://rxjs-dev.firebaseapp.com/api/operators/defaultIfEmpty) to provide a fallback value.
```

`ignoreNotFound(): (source: Observable<T>) => Observable<T>`

**Example**

```typescript
@Component({...})
export class AppComponent  {

>   user$: Observable<User | null>

Reviewer: ```suggestion
  user$: Observable<User>
```

  constructor(private http: HttpClient) {}

  ngOnInit() {
    this.user$ = this.http.get<User>('/users/123').pipe(

>       ignoreNotFound(),

Reviewer: ```suggestion
      ignoreNotFound()
```

>       defaultIfEmpty()

Reviewer: ```suggestion
```

I'd prefer the default example to show the operator on it's own, so people don't get confuse about the resulting type. People can refer to the docs of defaultIfEmpty when they still want a fallback value.
